### ISSUE TITLE
XML Route Document is complain that there are two routes

### ISSUE THREAD

**Issue by markejfmunro** [Vantiq/vantiq-extension-sources#374]
I have the folowing XML route document

```
<route>
  <from uri="vantiq://server.config"/>
  <to uri="paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883"/>
</route>
```

and I am getting the follwing error from the camel connector - it is complaining about multiple consumers but there is on to and one from!

PS C:\Data\ExtensionSource\31032023\camelConnector\camelConnector\bin> .\camelConnector
[Http http://localhost:8080/...] ebSocketListener#MQTTSourceXML INFO  WebSocket open
[Http http://localhost:8080/...] ebSocketListener#MQTTSourceXML INFO  Successful connection to MQTTSourceXML
[Http http://localhost:8080/...] CamelHandleConfiguration       DEBUG Fetching contents of document MQTTSink.xml from path /docs/MQTTSink.xml.
[Http http://localhost:8080/...] CamelRunner                    DEBUG Calling Run()
[Http http://localhost:8080/...] CamelRunner                    DEBUG AwaitStarted()
[                      Thread-1] MainSupport                    INFO  Apache Camel (Main) 3.18.3 is starting
[                      Thread-1] CamelRunner                    DEBUG >>>Using context: camel-1, stopped: true
[                      Thread-1] CamelRunner                    DEBUG Loading route from xml specificationType
[                      Thread-1] CamelDiscovery                 DEBUG Discovering Camel (version 3.18.3) components using context: camel-2
[                      Thread-1] EnumeratingComponentResolver   DEBUG Resolver asked to resolve component: vantiq
[                      Thread-1] PlaceholderEndpoint            DEBUG PlaceholderEndpoint for vantiq://server.config called
[                      Thread-1] EnumeratingDataFormatResolver  DEBUG Resolver asked to resolve component: jackson
[                      Thread-1] EnumeratingComponentResolver   DEBUG Resolver asked to resolve component: bean
[                      Thread-1] EnumeratingComponentResolver   DEBUG ... but this is a system provided entity, so we'll skip our resolution
[                      Thread-1] CamelDiscovery                 ERROR Trapped exception during component discovery:
org.apache.camel.FailedToCreateRouteException: Failed to create route route1 at: >>> Route(toRoute)[ -> [To[paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883]]] <<< in route: Route(route1)[From[vantiq://server.config] -> [ConvertBodyTo... because of Unsupported definition: Route(toRoute)[ -> [To[paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883]]]
        at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:241) ~[camel-core-reifier-3.18.3.jar:3.18.3]
        at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:75) ~[camel-core-reifier-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:862) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:750) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2947) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.support.service.BaseService.init(BaseService.java:83) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2630) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.support.service.BaseService.start(BaseService.java:111) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2649) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:262) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at io.vantiq.extsrc.camelconn.discover.CamelDiscovery.performComponentDiscovery(CamelDiscovery.java:75) ~[camelConnector.jar:?]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.constructClassLoader(CamelRunner.java:156) ~[camelConnector.jar:?]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.beforeStart(CamelRunner.java:369) ~[camelConnector.jar:?]
        at org.apache.camel.main.MainSupport.run(MainSupport.java:88) ~[camel-main-3.18.3.jar:3.18.3]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.lambda$runRoutes$0(CamelRunner.java:284) ~[camelConnector.jar:?]
        at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.IllegalStateException: Unsupported definition: Route(toRoute)[ -> [To[paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883]]]
        at org.apache.camel.reifier.ProcessorReifier.reifier(ProcessorReifier.java:173) ~[camel-core-reifier-3.18.3.jar:3.18.3]
        at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:229) ~[camel-core-reifier-3.18.3.jar:3.18.3]
        ... 16 more
[                      Thread-1] CamelDiscovery                 ERROR Trapped exception preparing or completing component discovery
io.vantiq.extsrc.camelconn.discover.DiscoveryException: Exception during component discovery
        at io.vantiq.extsrc.camelconn.discover.CamelDiscovery.performComponentDiscovery(CamelDiscovery.java:78) ~[camelConnector.jar:?]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.constructClassLoader(CamelRunner.java:156) ~[camelConnector.jar:?]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.beforeStart(CamelRunner.java:369) ~[camelConnector.jar:?]
        at org.apache.camel.main.MainSupport.run(MainSupport.java:88) ~[camel-main-3.18.3.jar:3.18.3]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.lambda$runRoutes$0(CamelRunner.java:284) ~[camelConnector.jar:?]
        at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: org.apache.camel.FailedToCreateRouteException: Failed to create route route1 at: >>> Route(toRoute)[ -> [To[paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883]]] <<< in route: Route(route1)[From[vantiq://server.config] -> [ConvertBodyTo... because of Unsupported definition: Route(toRoute)[ -> [To[paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883]]]
        at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:241) ~[camel-core-reifier-3.18.3.jar:3.18.3]
        at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:75) ~[camel-core-reifier-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:862) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:750) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2947) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.support.service.BaseService.init(BaseService.java:83) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2630) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.support.service.BaseService.start(BaseService.java:111) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2649) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:262) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at io.vantiq.extsrc.camelconn.discover.CamelDiscovery.performComponentDiscovery(CamelDiscovery.java:75) ~[camelConnector.jar:?]
        ... 5 more
Caused by: java.lang.IllegalStateException: Unsupported definition: Route(toRoute)[ -> [To[paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883]]]
        at org.apache.camel.reifier.ProcessorReifier.reifier(ProcessorReifier.java:173) ~[camel-core-reifier-3.18.3.jar:3.18.3]
        at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:229) ~[camel-core-reifier-3.18.3.jar:3.18.3]
        at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:75) ~[camel-core-reifier-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:862) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:750) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2947) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.support.service.BaseService.init(BaseService.java:83) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2630) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.support.service.BaseService.start(BaseService.java:111) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2649) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:262) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at io.vantiq.extsrc.camelconn.discover.CamelDiscovery.performComponentDiscovery(CamelDiscovery.java:75) ~[camelConnector.jar:?]
        ... 5 more
Exception in thread "Thread-1" java.lang.RuntimeException: run() in runRoutes failed
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.lambda$runRoutes$0(CamelRunner.java:286)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.RuntimeException: Unable to create route-based class loader
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.beforeStart(CamelRunner.java:371)
        at org.apache.camel.main.MainSupport.run(MainSupport.java:88)
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.lambda$runRoutes$0(CamelRunner.java:284)
        ... 1 more
Caused by: io.vantiq.extsrc.camelconn.discover.DiscoveryException: Exception preparing or completing component discovery processing
        at io.vantiq.extsrc.camelconn.discover.CamelDiscovery.performComponentDiscovery(CamelDiscovery.java:102)
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.constructClassLoader(CamelRunner.java:156)
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.beforeStart(CamelRunner.java:369)
        ... 3 more
Caused by: io.vantiq.extsrc.camelconn.discover.DiscoveryException: Exception during component discovery
        at io.vantiq.extsrc.camelconn.discover.CamelDiscovery.performComponentDiscovery(CamelDiscovery.java:78)
        ... 5 more
Caused by: org.apache.camel.FailedToCreateRouteException: Failed to create route route1 at: >>> Route(toRoute)[ -> [To[paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883]]] <<< in route: Route(route1)[From[vantiq://server.config] -> [ConvertBodyTo... because of Unsupported definition: Route(toRoute)[ -> [To[paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883]]]
        at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:241)
        at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:75)
        at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49)
        at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:862)
        at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:750)
        at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2947)
        at org.apache.camel.support.service.BaseService.init(BaseService.java:83)
        at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2630)
        at org.apache.camel.support.service.BaseService.start(BaseService.java:111)
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2649)
        at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:262)
        at io.vantiq.extsrc.camelconn.discover.CamelDiscovery.performComponentDiscovery(CamelDiscovery.java:75)
        ... 5 more
Caused by: java.lang.IllegalStateException: Unsupported definition: Route(toRoute)[ -> [To[paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883]]]
        at org.apache.camel.reifier.ProcessorReifier.reifier(ProcessorReifier.java:173)
        at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:229)
        ... 16 more
Terminate batch job (Y/N)?
^CThe syntax of the command is incorrect.
PS C:\Data\ExtensionSource\31032023\camelConnector\camelConnector\bin> ^C
PS C:\Data\ExtensionSource\31032023\camelConnector\camelConnector\bin> .\camelConnector
[Http http://localhost:8080/...] ebSocketListener#MQTTSourceXML INFO  WebSocket open
[Http http://localhost:8080/...] ebSocketListener#MQTTSourceXML INFO  Successful connection to MQTTSourceXML
[Http http://localhost:8080/...] CamelHandleConfiguration       DEBUG Fetching contents of document MQTTSink.xml from path /docs/MQTTSink.xml.
[Http http://localhost:8080/...] CamelRunner                    DEBUG Calling Run()
[Http http://localhost:8080/...] CamelRunner                    DEBUG AwaitStarted()
[                      Thread-1] MainSupport                    INFO  Apache Camel (Main) 3.18.3 is starting
[                      Thread-1] CamelRunner                    DEBUG >>>Using context: camel-1, stopped: true
[                      Thread-1] CamelRunner                    DEBUG Loading route from xml specificationType
[                      Thread-1] CamelDiscovery                 DEBUG Discovering Camel (version 3.18.3) components using context: camel-2
[                      Thread-1] EnumeratingComponentResolver   DEBUG Resolver asked to resolve component: vantiq
[                      Thread-1] PlaceholderEndpoint            DEBUG PlaceholderEndpoint for vantiq://server.config called
[                      Thread-1] EnumeratingComponentResolver   DEBUG Resolver asked to resolve component: paho
[                      Thread-1] PlaceholderEndpoint            DEBUG PlaceholderEndpoint for paho://CamelXML?brokerUrl=tcp://public.vantiq.com:1883 called
[                      Thread-1] AbstractCamelContext           INFO  Apache Camel 3.18.3 (camel-2) is starting
[                      Thread-1] AbstractCamelContext           INFO  Using 2 instances of same component class: io.vantiq.extsrc.camelconn.discover.PlaceholderComponent with names: paho, vantiq
[                      Thread-1] AbstractCamelContext           INFO  Routes startup (started:1)
[                      Thread-1] AbstractCamelContext           INFO      Started route1 (vantiq://server.config)
[                      Thread-1] AbstractCamelContext           INFO  Apache Camel 3.18.3 (camel-2) started in 107ms (build:2ms init:92ms start:13ms)
[                      Thread-1] CamelDiscovery                 DEBUG Component discovery complete:
[                      Thread-1] CamelDiscovery                 DEBUG     ---> paho
[                      Thread-1] CamelDiscovery                 DEBUG System components (ignored):
[                      Thread-1] CamelDiscovery                 DEBUG DataFormat discovery complete:
[                      Thread-1] CamelDiscovery                 DEBUG System dataformats (ignored):
[                      Thread-1] AbstractCamelContext           INFO  Apache Camel 3.18.3 (camel-2) is shutting down (timeout:45s)
[                      Thread-1] AbstractCamelContext           INFO  Routes stopped (stopped:1)
[                      Thread-1] AbstractCamelContext           INFO      Stopped route1 (vantiq://server.config)
[                      Thread-1] AbstractCamelContext           INFO  Apache Camel 3.18.3 (camel-2) shutdown in 13ms (uptime:0s)
:: loading settings :: url = jar:file:/C:/Data/ExtensionSource/31032023/camelConnector/camelConnector/lib/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[                      Thread-1] CamelResolver                  INFO  Resolving required libraries -- Component Resolution
[                      Thread-1] CamelResolver                  DEBUG CamelResolver MQTT, repos: [], cache: C:\Data\ExtensionSource\31032023\camelConnector\camelConnector\bin\componentCache, destination: C:\Data\ExtensionSource\31032023\camelConnector\camelConnector\bin\componentLib -- Making 9 artifacts available to Component Resolution
[                      Thread-1] VantiqEndpoint                 INFO  VantiqEndpoint creating consumer for uri: vantiq://localhost:8080 with sourceName: MQTTSourceXML, accessToken: present
[                      Thread-1] VantiqEndpoint                 INFO  VantiqEndpoint creating consumer for uri: vantiq://localhost:8080 with sourceName: MQTTSourceXML, accessToken: present
[                      Thread-1] AbstractCamelContext           INFO  Apache Camel 3.18.3 (camel-1) is starting
[                      Thread-1] AbstractCamelContext           ERROR Error starting CamelContext (camel-1) due to exception thrown: Failed to start route route1 because of Multiple consumers for the same endpoint is not allowed: vantiq://localhost:8080?sourceName=MQTTSourceXML&accessToken=xxxxxx&noSsl=true
org.apache.camel.FailedToStartRouteException: Failed to start route route1 because of Multiple consumers for the same endpoint is not allowed: vantiq://localhost:8080?sourceName=MQTTSourceXML&accessToken=xxxxxx&noSsl=true
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartOrResumeRouteConsumers(InternalRouteStartupManager.java:357) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartRouteConsumers(InternalRouteStartupManager.java:319) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.InternalRouteStartupManager.safelyStartRouteServices(InternalRouteStartupManager.java:213) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartOrResumeRoutes(InternalRouteStartupManager.java:147) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartCamel(AbstractCamelContext.java:3365) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartContext(AbstractCamelContext.java:3033) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.doStart(AbstractCamelContext.java:2988) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.support.service.BaseService.start(BaseService.java:119) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2649) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:262) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.doStart(CamelRunner.java:502) ~[camelConnector.jar:?]
        at org.apache.camel.support.service.BaseService.start(BaseService.java:119) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.main.MainSupport.run(MainSupport.java:89) ~[camel-main-3.18.3.jar:3.18.3]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.lambda$runRoutes$0(CamelRunner.java:284) ~[camelConnector.jar:?]
        at java.lang.Thread.run(Thread.java:833) [?:?]
[                      Thread-1] AbstractCamelContext           INFO  Apache Camel 3.18.3 (camel-1) is shutting down (timeout:45s)
[                      Thread-1] AbstractCamelContext           INFO  Routes stopped (stopped:2)
[                      Thread-1] AbstractCamelContext           INFO      Stopped route1 (vantiq://localhost:8080)
[                      Thread-1] AbstractCamelContext           INFO      Stopped route2 (vantiq://localhost:8080)
[                      Thread-1] AbstractCamelContext           INFO  Apache Camel 3.18.3 (camel-1) shutdown in 6ms (uptime:1s)
[                      Thread-1] CamelRunner                    ERROR Camel startup has failed due to:
org.apache.camel.FailedToStartRouteException: Failed to start route route1 because of Multiple consumers for the same endpoint is not allowed: vantiq://localhost:8080?sourceName=MQTTSourceXML&accessToken=xxxxxx&noSsl=true
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartOrResumeRouteConsumers(InternalRouteStartupManager.java:357) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartRouteConsumers(InternalRouteStartupManager.java:319) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.InternalRouteStartupManager.safelyStartRouteServices(InternalRouteStartupManager.java:213) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartOrResumeRoutes(InternalRouteStartupManager.java:147) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartCamel(AbstractCamelContext.java:3365) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartContext(AbstractCamelContext.java:3033) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.doStart(AbstractCamelContext.java:2988) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.support.service.BaseService.start(BaseService.java:119) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2649) ~[camel-base-engine-3.18.3.jar:3.18.3]
        at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:262) ~[camel-core-engine-3.18.3.jar:3.18.3]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.doStart(CamelRunner.java:502) ~[camelConnector.jar:?]
        at org.apache.camel.support.service.BaseService.start(BaseService.java:119) ~[camel-api-3.18.3.jar:3.18.3]
        at org.apache.camel.main.MainSupport.run(MainSupport.java:89) ~[camel-main-3.18.3.jar:3.18.3]
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.lambda$runRoutes$0(CamelRunner.java:284) ~[camelConnector.jar:?]
        at java.lang.Thread.run(Thread.java:833) [?:?]
[                      Thread-1] CamelRunner                    DEBUG NotifyStarted()
[                      Thread-1] CamelRunner                    DEBUG NotifyStarted() completed
[Http http://localhost:8080/...] CamelRunner                    DEBUG AwaitStarted() completed
Exception in thread "Thread-1" java.lang.RuntimeException: run() in runRoutes failed
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.lambda$runRoutes$0(CamelRunner.java:286)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.camel.FailedToStartRouteException: Failed to start route route1 because of Multiple consumers for the same endpoint is not allowed: vantiq://localhost:8080?sourceName=MQTTSourceXML&accessToken=xxxxxx&noSsl=true
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartOrResumeRouteConsumers(InternalRouteStartupManager.java:357)
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartRouteConsumers(InternalRouteStartupManager.java:319)
        at org.apache.camel.impl.engine.InternalRouteStartupManager.safelyStartRouteServices(InternalRouteStartupManager.java:213)
        at org.apache.camel.impl.engine.InternalRouteStartupManager.doStartOrResumeRoutes(InternalRouteStartupManager.java:147)
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartCamel(AbstractCamelContext.java:3365)
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartContext(AbstractCamelContext.java:3033)
        at org.apache.camel.impl.engine.AbstractCamelContext.doStart(AbstractCamelContext.java:2988)
        at org.apache.camel.support.service.BaseService.start(BaseService.java:119)
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2649)
        at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:262)
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.doStart(CamelRunner.java:502)
        at org.apache.camel.support.service.BaseService.start(BaseService.java:119)
        at org.apache.camel.main.MainSupport.run(MainSupport.java:89)
        at io.vantiq.extsrc.camelconn.discover.CamelRunner.lambda$runRoutes$0(CamelRunner.java:284)
        ... 1 more

**Comment by fhcarter** [comment]
@markejfmunro, the XML route definition is missing from the report.  So I'll need a bit more data :-)

**Comment by fhcarter** [comment]
@markejfmunro 

Never mind -- the issue was that the XML wasn't escaped, so GIT ate it.  I escaped it.  Will get to this soon (probably early next week)

**Comment by fhcarter** [comment]
@markejfmunro,

The problem here appears to be a Camel problem, though it's certainly possible that the connector and/or component are doing something that violates some understanding somewhere.  However, I found Camel issues [14969](https://issues.apache.org/jira/browse/CAMEL-14969) and [13651](https://issues.apache.org/jira/browse/CAMEL-13651) which appear to be similar but not quite the same. They are marked _resolved_, but it certainly appears that there's something similar going on..

Anyway, what appears to be the _correct_ thing to do here is to add an id to your route.  When I change the route to 

```
<route id="issue374">
    <from uri="vantiq://server.config"/>
    <to uri="paho:CamelXML?brokerUrl=tcp://public.vantiq.com:1883"/>
</route>
```

Things operate as one would expect (at least as far as I can tell -- Camel starts up). You are certain encouraged to use a more meaningful route id :-).

I'll close this issue (probably Monday) by adding an error to the `VantiqEndpoint.createConsumer()` code stating:

> More than one consumer created for this endpoint. Check that this endpoint's consumer is
 not used in multiple places. That said, sometimes this error result from defining
routes without ids.  Please add ids to any routes in use and retry.

to help move folks in a workaround direction.  Once we log that error, we operate as we do now and Camel will complain vociferously, as is its way. The change I'll make has no other changes -- it detects the situation, notes it, but otherwise operates as it did.  So you should be fine adding the id to your route and attempting to move forward.

Thanks for the detailed report.  It certainly makes finding things easier.

**Comment by fhcarter** [comment]
Resolution: Camel issue with routes that lack ids.  Adding an id to the route, _e.g.,_

```
```

solves the problem.

Will close issue by logging an error when we detect multiple consumers on the same endpoint.  This will hopefully point callers in the right direction.

**Comment by fhcarter** [comment]
Fixed via #375
